### PR TITLE
improve exception message when no identifier are defined

### PR DESCRIPTION
**Pull request by ronfroy**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

### CODE REVIEW

**Comment by teohhanhui** [comment]
I'm -1 as error messages shouldn't be verbose. We should communicate clearly what's the error encountered, but nothing else.

**Comment by ronfroy** [comment]
@teohhanhui it saves time to have a useful exception message

**Comment by teohhanhui** [comment]
If we want to give a better error message for the case when there are no identifiers, we should add a check after

https://github.com/api-platform/core/blob/0a69709067335fb2ccb5bd48b0850b47f3aaf72d/src/Bridge/Symfony/Routing/IriConverter.php#L118

maybe.

**Comment by teohhanhui** [comment]
It also means we don't need to say "maybe". :smile:

**Comment by ronfroy** [comment]
@teohhanhui done

**Comment by teohhanhui** [comment]
That was not the right way to fix it. :smile:

Just add a check at the beginning of `generateIdentifiersUrl` that there are indeed any identifiers at all:

https://github.com/api-platform/core/blob/0a69709067335fb2ccb5bd48b0850b47f3aaf72d/src/Bridge/Symfony/Routing/IriConverter.php#L177-L179

And I think we could still target `2.2` since it could be considered a bug fix, and `2.3` is not released yet. :laughing:

**Comment by ronfroy** [comment]
done

File: src/Bridge/Symfony/Routing/IriConverter.php [hunk 1]
```diff
@@ -170,12 +170,22 @@ public function getSubresourceIriFromResourceClass(string $resourceClass, array
     /**
      * Generate the identifier url.
      *
-     * @param array $identifiers
+     * @param array  $identifiers
+     * @param object $resourceClass
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Routing/IriConverter.php]: `$resourceClass` is a `string`. You should also add the type hint for the parameter.

**Comment by teohhanhui** [comment]
Can you change this to target `2.2`?

**Pull request closed without merging**